django/db/backends/utils.py

```diff
@@ -178,13 +179,19 @@ def rev_typecast_decimal(d):
 
 
 def truncate_name(name, length=None, hash_len=4):
-    """Shortens a string to a repeatable mangled version with the given length.
     """
-    if length is None or len(name) <= length:
+    Shortens a string to a repeatable mangled version with the given length.
+    If quote striped name contains username e.g. USERNAME"."TABLE, use name
+    without username.
```

Comment:
I'm not sure what "use name without username" means. It seems to suggest USERNAME"."TABLE because become TABLE but I don't think that's the behavior.